src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java

```diff
@@ -95,59 +91,129 @@ public void testStructuredDataType() throws Exception {
+            + "_x27_x44_xFE_x0F binary, "
+            + "\"\"\"\" date, "
+            + "\"\\\" string, "
+            + "\"\" string, "
+            + "\"временнаяметка समयमोहर 時間戳記 ㄕˊㄔㄨㄛ 타임스탬프 タイムスタンプ tidsstämpel\" string, "
+            + "\".\" map(string, int),"
+            + "\"\\.\" array(int),"
+            + "obj object(obj boolean))",
+        val,
+        "select {columnName} from {tableName}",
+        val);
+
+    /* Multiple rows test, check parquet file */
+    val =
+        (Iterable<Object>)
+            objectMapper.readValue(
+                "[{\"obj\\.obj\": false, \"test_test\": 1, \"test_x5Ftest\": 2, \"obj\\\\.obj\":"
+                    + " 3.0, \"❄️\": 4.0, \"5566\": \"5.0\", \"_5566\": \"6.0\", \"_\":"
+                    + " \"41424344454647484950\", \"_x27_x44_xFE_x0F\": \"41424344\", \"\\\"\":"
+                    + " \"2024-01-01\", \"\\\\\": \"12:00:00\", \"\":"
+                    + " \"2024-01-01T12:00:00.000000\", \"временнаяметка समयमोहर 時間戳記 ㄕˊㄔㄨㄛ 타임스탬프"
+                    + " タイムスタンプ tidsstämpel\": \"2024-01-01T12:00:00.000000+08:00\", \"\\.\":"
+                    + " {\"key1\": 1}, \"\\\\.\": [1, 2, 3], \"obj\": {\"obj\":"
+                    + " true}},{\"obj\\.obj\": true, \"test_test\": 2, \"test_x5Ftest\": 3,"
+                    + " \"obj\\\\.obj\": 4.0, \"❄️\": 5.0, \"5566\": \"6.0\", \"_5566\": \"7.0\","
+                    + " \"_\": \"51525354555657585960\", \"_x27_x44_xFE_x0F\": \"51525354\","
+                    + " \"\\\"\": \"2024-01-02\", \"\\\\\": \"13:00:00\", \"\":"
+                    + " \"2024-01-02T13:00:00.000000\", \"временнаяметка समयमोहर 時間戳記 ㄕˊㄔㄨㄛ 타임스탬프"
+                    + " タイムスタンプ tidsstämpel\": \"2024-01-02T13:00:00.000000+08:00\", \"\\.\":"
+                    + " {\"key2\": 2}, \"\\\\.\": [4, 5, 6], \"obj\": {\"obj\":"
+                    + " false}},{\"obj\\.obj\": false, \"test_test\": 3, \"test_x5Ftest\": 4,"
+                    + " \"obj\\\\.obj\": 5.0, \"❄️\": 6.0, \"5566\": \"7.0\", \"_5566\": \"8.0\","
+                    + " \"_\": \"61626364656667686970\", \"_x27_x44_xFE_x0F\": \"61626364\","
+                    + " \"\\\"\": \"2024-01-03\", \"\\\\\": \"14:00:00\", \"\":"
+                    + " \"2024-01-03T14:00:00.000000\", \"временнаяметка समयमोहर 時間戳記 ㄕˊㄔㄨㄛ 타임스탬프"
+                    + " タイムスタンプ tidsstämpel\": \"2024-01-03T14:00:00.000000+08:00\", \"\\.\":"
+                    + " {\"key3\": 3}, \"\\\\.\": [7, 8, 9], \"obj\": {\"obj\": true}}]",
+                Object.class);
+
+    testIcebergIngestAndQuery(
+        "object("
+            + "\"obj.obj\" boolean, "
+            + "test_test int, "
+            + "test_x5Ftest long, "
+            + "\"obj\\.obj\" float, "
+            + "\"❄️\" double, "
+            + "\"5566\" string, "
+            + "_5566 string, "
+            + "\"_\" fixed(10), "
+            + "_x27_x44_xFE_x0F binary, "
+            + "\"\"\"\" date, "
+            + "\"\\\" string, "
```

Comment:
Keep it as String for now as we cannot compare date time value without format. Planning to add a generator & comparator for all types / structured data types.